snakemake_rules/Snakefile

```diff
@@ -0,0 +1,134 @@
+    shell:
+        "fastqc --threads {threads} --noextract --dir {params.tmp_dir} --outdir {params.outpath} {input.read1};"
+        "fastqc --threads {threads} --noextract --dir {params.tmp_dir} --outdir {params.outpath} {input.read2};"
+
+rule trim_galore_pe:
+    input:
+        read1 = lambda wildcards: config["read1"][wildcards.sample],
+        read2 = lambda wildcards: config["read2"][wildcards.sample]
+    output:
+        trimmed_1 = COMMON_DIR + "/{sample}/trim-galore/{sample}_1_val_1.fq.gz",
+        trimmed_2 = COMMON_DIR + "/{sample}/trim-galore/{sample}_2_val_2.fq.gz"
+    params:
+        outpath = COMMON_DIR + "/{sample}/trim-galore/"
+    shell:
+         "trim_galore --cores 15 --fastqc --gzip --output_dir {params.outpath} --paired {input.read1} {input.read2}"
+
+rule star_index:
+    input:
+         fa = config["genome"],
+         gtf = config["gtf"]
+    output:
+         directory(COMMON_DIR + "/star_index/")
+    threads: 36
+    shell:
+         'STAR --runThreadN {threads} '
+         '--runMode genomeGenerate '
+         '--genomeDir {output}'
+         '--genomeFastaFiles {input.fa} '
+         '--sjdbGTFfile {input.gtf} '
+         '--sjdbOverhang 99'
+
+rule star_align:
+    input:
+        trimmed_1 = COMMON_DIR + "/{sample}/trim-galore/{sample}_1_val_1.fq.gz",
+        trimmed_2 = COMMON_DIR + "/{sample}/trim-galore/{sample}_2_val_2.fq.gz",
+        ref_dir = COMMON_DIR + "/star_index/"
+    output:
+        COMMON_DIR + "/{sample}/star_aln/{sample}.Aligned.sortedByCoord.out.bam",
+        COMMON_DIR + "/{sample}/star_aln/{sample}.Aligned.toTranscriptome.out.bam"
+    params:
+        prefix = COMMON_DIR + "/{sample}/star_aln/{sample}."
+    log:
+        COMMON_DIR + "/{sample}/star_aln/logs/{sample}.log",
+    threads: 36
+    shell:
+        "STAR --genomeDIR {input.ref_dir} "
+        "--readFilesin {input.trimmed_1} {input.trimmed_2} "
+        "--readFilesCommand zcat "        
+        "--twopassMode Basic "
+        "--quantMode TranscriptomeSAM " # using RSEM
+        "--peOverlapNbasesMin 10 "
+        "--outSAMtype BAM SortedByCoordinate "
+        "--outFileNamePrefix {params.prefix} "
+        "--runThreadN {threads}"
+
+
+# rule rename_bam: # I will add this rule .. later to comply w/ conventions ^_^
+
+rule mark_dupes:
+    input:
+        COMMON_DIR + "/{sample}/star_aln/{sample}.Aligned.sortedByCoord.out.bam"
+    output:
+        bam=COMMON_DIR + "/{sample}/picard_markdupe/{sample}.bam",
+        metrics=COMMON_DIR + "/{sample}/picard_markdupe/{sample}.metrics.txt"
+    log:
+        COMMON_DIR + "/{sample}/logs/{sample}.log"
+    params:
+        "REMOVE_DUPLICATES=true"
+    wrapper:
+        "0.68.0/bio/picard/markduplicates"
+
+rule samtools_index:
+    input:
+        COMMON_DIR + "/{sample}/star_aln/{sample}.Aligned.sortedByCoord.out.bam"
+    output:
+        COMMON_DIR + "/{sample}/star_aln/{sample}.Aligned.sortedByCoord.out.bam.bai"
+    wrapper:
+        "0.68.0/bio/samtools/index"
```

Comment:
using wrapper here b/c why not 🤷‍♀️ 

@jemten should I also index `.Aligned.toTranscriptome.out.bam"`

Reply:
Why not 😄